pinot-tools/src/main/java/com/linkedin/pinot/tools/AutoAddInvertedIndex.java

```diff
@@ -0,0 +1,212 @@
+import org.json.JSONObject;
+
+
+/**
+ * This tool will directly modify the table configs in ZooKeeper.
+ * PLEASE USE IT VERY CAREFULLY!
+ */
+public class AutoAddInvertedIndex {
+  private AutoAddInvertedIndex() {
+  }
+
+  private static final String AUTO_GENERATED_INVERTED_INDEX_KEY = "autoGeneratedInvertedIndex";
+
+  private static final String ZK_ADDRESS = "zkAddress:zkPort";
+  private static final String CLUSTER_NAME = "pinot-cluster";
+  private static final String TABLE_CONFIG_PATH = "/CONFIGS/TABLE/";
+  private static final String SCHEMA_PATH = "/SCHEMAS/";
+  private static final ZKHelixAdmin HELIX_ADMIN = new ZKHelixAdmin(ZK_ADDRESS);
+  private static final ZkHelixPropertyStore<ZNRecord> PROPERTY_STORE =
+      new ZkHelixPropertyStore<>(ZK_ADDRESS, new ZNRecordSerializer(),
+          PropertyPathConfig.getPath(PropertyType.PROPERTYSTORE, CLUSTER_NAME));
+
+  private static final String BROKER_QUERY_URL = "http://brokerAddress:brokerPort/query";
+
+  private static final long TABLE_SIZE_THRESHOLD = 100_000_000;
+
+  private static AbstractTableConfig getTableConfig(String tableName)
+      throws Exception {
+    return AbstractTableConfig.fromZnRecord(PROPERTY_STORE.get(TABLE_CONFIG_PATH + tableName, null, 0));
+  }
+
+  private static void setTableConfig(String tableName, AbstractTableConfig tableConfig)
+      throws Exception {
+    ZNRecord znRecord = AbstractTableConfig.toZnRecord(tableConfig);
+    znRecord.setSimpleField(AUTO_GENERATED_INVERTED_INDEX_KEY, "true");
+    PROPERTY_STORE.set(TABLE_CONFIG_PATH + tableName, znRecord, 1);
+  }
+
+  private static Schema getTableSchema(String tableName)
+      throws Exception {
+    ZNRecord znRecord = PROPERTY_STORE.get(SCHEMA_PATH + tableName, null, 0);
+    if (znRecord == null) {
+      return null;
+    } else {
+      return SchemaUtils.fromZNRecord(znRecord);
+    }
+  }
+
+  private static JSONObject sendQuery(String query)
+      throws Exception {
+    URLConnection urlConnection = new URL(BROKER_QUERY_URL).openConnection();
+    urlConnection.setDoOutput(true);
+
+    BufferedWriter writer = new BufferedWriter(new OutputStreamWriter(urlConnection.getOutputStream(), "UTF-8"));
+    writer.write(new JSONObject().put("pql", query).toString());
+    writer.flush();
+
+    BufferedReader reader = new BufferedReader(new InputStreamReader(urlConnection.getInputStream(), "UTF-8"));
+    return new JSONObject(reader.readLine());
+  }
+
+  private static class ResultPair implements Comparable<ResultPair> {
+    private final String _key;
+    private final long _value;
+
+    public ResultPair(String key, long value) {
+      _key = key;
+      _value = value;
+    }
+
+    @Override
+    public int compareTo(@Nonnull ResultPair o) {
+      return Long.compare(o._value, _value);
+    }
+
+    @Override
+    public String toString() {
+      return _key + ": " + _value;
+    }
+  }
+
+  public static void main(String[] args)
```

Comment:
Invoke it through pinot-admin so that you get args parsing for free? You need to take zk path and cluster name as argument, I suppose.